docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,167 @@
+---
+title: 'Overlay'
+slug: /engines/database-engines/overlay
+description: 'Creates a logical read-only database that exposes the union of tables from multiple underlying databases.'
+sidebar_label: 'Overlay'
+sidebar_position: 51
+---
+
+# `DatabaseOverlay` — design & behavior (current implementation)
+
+## What is `DatabaseOverlay`? {#introduction}
+
+`DatabaseOverlay` is a logical “view” database that exposes **the union of tables from multiple underlying databases**. It does **not** own data itself. It supports two modes:
+
+* `Mode::OwnedMembers` — used by **clickhouse-local** helper code; the overlay sits *on top of* member DBs it creates/owns.
+* `Mode::FacadeOverCatalog` — used by the SQL factory (`ENGINE = Overlay(...)`); the overlay is a **read-only facade** over existing, separately managed databases.
+
+---
+
+## Construction {#construction}
+
+### Factory (SQL) {#sql}
+
+```sql
+CREATE DATABASE dboverlay ENGINE = Overlay('db_overlay_a', 'db_overlay_b');
+```
+
+* The factory resolves every underlying DB name via `DatabaseCatalog::tryGetDatabase(name)`.
```

Comment:
```suggestion
* The factory resolves every underlying database name via `DatabaseCatalog::tryGetDatabase(name)`.
```

(here and below)

Reply:
This is kind of developers guide, let's move this under `## Developers guide` here or maybe you have better ideas.

Reply:
I have a better idea, let's split it into a tiny description and use case example of what the overlay database is, and a separate tech guide, should be in a separate file (maybe under `development`, but need to make sure that it will be rendered somehow, or maybe just a `DatabaseOverlay.md` under `src/Databases`, or even a comment (if this description can be trimmed somehow) above the `class DatabaseOverlay` - this will be the most suitable place I think, since I would say that it is more likely that developers will read the file then find this documentation)